corehq/apps/users/forms.py

```diff
@@ -81,7 +80,10 @@ def clean_mobile_worker_username(domain, username, name_too_long_message=None,
     username = format_username(username, domain)
     validate_username(username)
 
-    if CouchUser.username_exists(username):
+    exists = user_exists(username)
+    if exists.exists:
+        if exists.is_deleted:
+            raise forms.ValidationError(_('This username was used previously.'))
```

Comment:
This should never happen since the UI blocks the action but I've added it as a safeguard.